### ISSUE TITLE
Prune transaction metadata while pruning

### ISSUE THREAD

**Issue by GalRogozinski** [iotaledger/iri#1307]
### Description
We store in the db a mapping between addresses and transaction hashes that spent/deposited iotas to the address.

When the pruning of transactions occur, we still keep the mapping of addresses to transactions.
We should delete the transaction hashes from the mapping that were pruned.

### Motivation
We load a huge number of irrelevant hashes that may cause OOM errors. This new feature should relieve the problem.

### Requirements
1. Once we are done with a pruning operation we should delete all the pruned transactions' hashes from the `AddressViewModel` db.

**Comment by jakubcech** [comment]
✔️ by Trinity.

**Comment by ovanwijk** [comment]
I agree with this as well! We should also consider the BundleViewModel, TagViewModel and AproveeViewModel. These are similar indexes.

As an idea, to prevent more costly constructions of the 'Hashes' class and loading into a set to delete it I already implemented an efficient method for just cutting out the 'indexable' part from the saved set on a separate branch.

https://github.com/ovanwijk/iri/blob/d69046319ed2656a2ef7f1f4bf03eaf2598ca517/src/main/java/com/iota/iri/storage/rocksDB/RocksDBPPPImpl.java#L311-L338

**Comment by jakubcech** [comment]
We should break this down and include the different models, as it's already quite an expensive task at this point. 

>As an idea, to prevent more costly constructions of the 'Hashes' class and loading into a set to delete it I already implemented an efficient method for just cutting out the 'indexable' part from the saved set on a separate branch.

👍
